newIDE/app/src/Utils/GDevelopServices/Project.js

```diff
@@ -0,0 +1,271 @@
+    `${GDevelopProjectApi.baseUrl}/project/${cloudProjectId}/action/commit`,
+    { newVersion },
+    {
+      headers: {
+        Authorization: authorizationHeader,
+      },
+      params: { userId },
+    }
+  );
+  return response.data;
+};
+
+export const listUserCloudProjects = async (
+  getAuthorizationHeader: () => Promise<string>,
+  userId: string
+): Promise<Array<CloudProjectWithUserAccessInfo>> => {
+  const authorizationHeader = await getAuthorizationHeader();
+  const response = await axios.get(`${GDevelopProjectApi.baseUrl}/project`, {
+    headers: { Authorization: authorizationHeader },
+    params: { userId },
+  });
+  return response.data;
+};
+
+export const getCloudProject = async (
+  authenticatedUser: AuthenticatedUser,
+  cloudProjectId: string
+): Promise<?CloudProject> => {
+  const { getAuthorizationHeader, firebaseUser } = authenticatedUser;
+  if (!firebaseUser) return;
+
+  const { uid: userId } = firebaseUser;
+  const authorizationHeader = await getAuthorizationHeader();
+  const response = await axios.get(
+    `${GDevelopProjectApi.baseUrl}/project/${cloudProjectId}`,
+    {
+      headers: {
+        Authorization: authorizationHeader,
+      },
+      params: { userId },
+    }
+  );
+  return response.data;
+};
+
+export const updateCloudProject = async (
+  authenticatedUser: AuthenticatedUser,
+  cloudProjectId: string,
+  attributes: { name: string }
+): Promise<?CloudProject> => {
+  const { getAuthorizationHeader, firebaseUser } = authenticatedUser;
+  if (!firebaseUser) return;
+
+  const cleanedAttributes = {
+    name: attributes.name.slice(0, CLOUD_PROJECT_NAME_MAX_LENGTH),
+  };
+
+  const { uid: userId } = firebaseUser;
+  const authorizationHeader = await getAuthorizationHeader();
+  const response = await axios.patch(
+    `${GDevelopProjectApi.baseUrl}/project/${cloudProjectId}`,
+    cleanedAttributes,
+    {
+      headers: {
+        Authorization: authorizationHeader,
+      },
+      params: { userId },
+    }
+  );
+  return response.data;
+};
+
+export const deleteCloudProject = async (
+  authenticatedUser: AuthenticatedUser,
+  cloudProjectId: string
+): Promise<?CloudProject> => {
+  const { getAuthorizationHeader, firebaseUser } = authenticatedUser;
+  if (!firebaseUser) return;
+
+  const { uid: userId } = firebaseUser;
+  const authorizationHeader = await getAuthorizationHeader();
+  const response = await axios.delete(
+    `${GDevelopProjectApi.baseUrl}/project/${cloudProjectId}`,
+    {
+      headers: {
+        Authorization: authorizationHeader,
+      },
+      params: { userId },
+    }
+  );
+  return response.data;
+};
+
+export const getPresignedUrlForVersionUpload = async (
+  authenticatedUser: AuthenticatedUser,
+  cloudProjectId: string
+): Promise<string> => {
+  const { getAuthorizationHeader, firebaseUser } = authenticatedUser;
+  if (!firebaseUser)
+    throw new Error(
+      'Presigned url is could not be created, user not authenticated.'
```

Comment:
`is` can be removed

Reply:
by the way, this is the only place where you throw an error, you could probably just return;